app/src/test/java/com/codesoom/assignment/controllers/TaskControllerTest.java

```diff
@@ -0,0 +1,247 @@
+import com.codesoom.assignment.models.Task;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@DisplayName("TaskController class")
+class TaskControllerTest {
+
+    private final Long ID = 1L;
+    private final String TITLE = "SAMPLE TITLE";
+    private final String UPDATED = "[UPDATED]";
+
+    private TaskController controller;
+    private TaskService taskService;
+    private Task task;
+
+    @BeforeEach
+    void setUp() {
+        taskService = new TaskService();
+        controller = new TaskController(taskService);
+
+        task = new Task();
+        task.setId(ID);
+        task.setTitle(TITLE);
+
+        controller.create(task);
+    }
+
+    @Nested
+    @DisplayName("list method")
+    class DescribeGetTasks {
+        @Nested
+        @DisplayName("when exist tasks")
+        class ContextWithTasks {
+            @Test
+            @DisplayName("It returns the list contains tasks")
+            void list() {
+                List<Task> tasks = controller.list();
+                assertThat(tasks).isNotEmpty();
+                assertThat(tasks).hasSize(1);
+            }
+        }
+
+        @Nested
+        @DisplayName("when empty tasks")
+        class ContextWithEmptyTasks {
+            @BeforeEach
+            void prepared() {
+                controller.delete(ID);
+            }
+
+            @Test
+            @DisplayName("It returns the empty list")
+            void list() {
+                List<Task> tasks = controller.list();
+                assertThat(tasks).isEmpty();
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("detail method")
+    class DescribeGetTask {
+        @Nested
+        @DisplayName("when exist matched task")
+        class ContextWithMatchedTask {
+            @Test
+            @DisplayName("It returns a task that matches the ID")
+            void detailTheTask() {
+                Task task = controller.detail(ID);
+
+                assertThat(task).isNotNull();
+                assertThat(task.getId()).isEqualTo(ID);
+                assertThat(task.getTitle()).isEqualTo(TITLE);
+            }
+        }
+
+        @Nested
+        @DisplayName("when no matched task")
+        class ContextWithNoMatchedTask {
+            @Test
+            @DisplayName("It throw the task not found exception")
+            void detailNoTask() {
+                assertThatThrownBy(() -> controller.detail(ID * 0L))
+                        .isInstanceOf(TaskNotFoundException.class);
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("create method")
+    class DescribeCreateTask {
+        @Nested
+        @DisplayName("when a valid task")
+        class ContextWithValidTask {
```

Comment:
이 컨텍스트에서는 `when`보다 `with`가 적합해 보입니다.
```suggestion
        @Nested
        @DisplayName("with a valid task")
        class ContextWithValidTask {
```
그리고 `valid task`를 이 스코프에서 정의해보세요.